src/cython/vapoursynth.pyx

```diff
@@ -1225,6 +1267,47 @@ cdef FrameProps createFrameProps(RawFrame f):
 Mapping.register(FrameProps)
 
 
+cdef class ChannelLayout(int):
+    cdef readonly int num_channels
+
+    def __init__(self):
+        raise Error('Class cannot be instantiated directly')
+
+    def __contains__(self, layout):
+        return bool(self & (1 << layout))
+
+    def __iter__(self):
+        for v in AudioChannels:
+            if ((1 << v) & self):
+                yield v
+
+    def __eq__(self, other):
+        if not isinstance(other, ChannelLayout):
+            return False
+        return other == self
+
+    def __len__(self):
+        return self.num_channels
```

Comment:
I think, `self.bit_count()` would be another solution that doesn't require `num_channels`.

Reply:
All right, so we only use the channel layout and can put this on public API